azext_iot/tests/iothub/message_endpoint/test_iothub_message_endpoint_int.py

```diff
@@ -0,0 +1,906 @@
+            iot_hub,
+            iot_rg,
+            endpoint_names[2],
+            iot_rg,
+            endpoint_uri,
+            eventhub_instance,
+            user_id
+        )
+    )
+
+    expected_user_endpoint = build_expected_endpoint(
+        endpoint_names[2],
+        iot_rg,
+        iot_sub,
+        entity_path=eventhub_instance,
+        authentication_type=AuthenticationType.IdentityBased.value,
+        endpoint_uri=endpoint_uri,
+        identity=user_id
+    )
+
+    endpoint_output = cli.invoke(
+        "iot hub message-endpoint show -n {} -g {} --en {}".format(
+            iot_hub, iot_rg, endpoint_names[2]
+        )
+    )
+
+    assert_endpoint_properties(endpoint_output.as_json(), expected_user_endpoint)
+
+    # List
+
+    endpoint_list = cli.invoke(
+        "iot hub message-endpoint list -n {} -g {}".format(
+            iot_hub, iot_rg
+        )
+    ).as_json()
+
+    eventhub_list = cli.invoke(
+        "iot hub message-endpoint list -n {} -g {} -t {}".format(
+            iot_hub, iot_rg, "eventhub"
+        )
+    ).as_json()
+
+    assert len(eventhub_list) == 3
+    assert endpoint_list["eventHubs"] == eventhub_list
+
+    # Delete one event hub endpoint
+    cli.invoke(
+        "iot hub message-endpoint delete -n {} -g {} --en {} -y".format(
+            iot_hub, iot_rg, endpoint_names[0]
+        )
+    )
+
+    # ensure that only one got deleted
+    eventhub_list = cli.invoke(
+        "iot hub message-endpoint list -n {} -g {} -t {}".format(
+            iot_hub, iot_rg, "eventhub"
+        )
+    ).as_json()
+
+    assert len(eventhub_list) == 2
+
+    # Delete all event hub endpoints
+    cli.invoke(
+        "iot hub message-endpoint delete -n {} -g {} -t {} -y".format(
+            iot_hub, iot_rg, "eventhub"
+        )
+    )
+
+    endpoint_list = cli.invoke(
+        "iot hub message-endpoint list -n {} -g {} -t {}".format(
+            iot_hub, iot_rg, "eventhub"
+        )
+    ).as_json()
+
+    assert endpoint_list == []
+
+
+def test_iot_servicebus_endpoint_lifecycle(provisioned_service_bus_with_identity_module):
+    # this test covers two endpoint types
+    iot_hub_obj, servicebus_obj = provisioned_service_bus_with_identity_module
+    iot_hub = iot_hub_obj["name"]
+    iot_rg = iot_hub_obj["resourcegroup"]
+    iot_sub = iot_hub_obj["subscriptionid"]
+    user_id = list(iot_hub_obj["identity"]["userAssignedIdentities"].keys())[0]
+
+    queue_instance = servicebus_obj["queue"]["name"]
+    topic_instance = servicebus_obj["topic"]["name"]
+    endpoint_uri = "sb:" + servicebus_obj["namespace"]["serviceBusEndpoint"].split(":")[1]
+    queue_cs = servicebus_obj["queueConnectionString"]
+    topic_cs = servicebus_obj["topicConnectionString"]
+    # create 6 names, 2 types of service bus endpoints * 3 auth types
+    endpoint_names = generate_ep_names(6)
+
+    # create topic endpoint first - connection string
+    cli.invoke(
```

Comment:
we should have checks for invalid commands

Reply:
oh but this is the embedded cli so the error messages will litter the logs - would we still want these

Reply:
will omit for now